util/src/main/java/com/farao_community/farao/util/AbstractNetworkPool.java

```diff
@@ -24,6 +25,9 @@ public abstract class AbstractNetworkPool extends ForkJoinPool implements AutoCl
     // State used to save initial content of target variant.
     // Useful when targetVariant equals VariantManagerConstants.INITIAL_VARIANT_ID
     protected final String stateSaveVariant;
+    protected Network network;
+    protected String networkInitialVariantId;
+    protected Set<String> baseNetworkVariantIds;
 
     public static AbstractNetworkPool create(Network network, String targetVariant, int parallelism) {
```

Comment:
how about adding a 4th boolean argument "initClones". if set to true, the clones would be added in constructor (used in all classes except search tree). if set to false, then the user of the class should call addClones explicitely. This will at least make it lighter for classes other than search tree